python/ray/autoscaler/_private/aws/config.py

```diff
@@ -107,89 +100,105 @@ def log_to_cli(config):
     cli_logger.doassert(provider_name is not None,
                         "Could not find a pretty name for the AWS provider.")
 
+    head_node_type = config["head_node_type"]
+    head_node_config = config["available_node_types"][head_node_type][
+        "node_config"]
+
     with cli_logger.group("{} config", provider_name):
 
         def same_everywhere(key):
```

Comment:
what does this function do? Can we add type annotation and description please?

Reply:
Maybe an improvement can be (check if the the KeyName is the same in the node configs of all available node types)